app.js

```diff
@@ -246,3 +262,17 @@ app.get('/', function(req, res) {
 });
 
 
+function infiniteLoop() {
+    /* this will launch other scheduled tasks too */
+    return Promise
+        .resolve()
+        .delay(60 * 1000)
+        .then(function() {
+            if(_.size(performa.queue))
+                return mongo
+                    .cacheFlush(performa.queue, "performa")
+        })
+        .then(infiniteLoop);
```

Comment:
OK, it's not recursive, probably because promises are just registered in the event loop without adding a new frame to the stack. This code doesn't segfault:

```javascript
var Promise = require('bluebird');

function rec (i) {
    return Promise
        .resolve(i + 1)
        .tap(console.log(i))
        .then(rec);
}

rec(0);
```

Anyway, I still think a `setInterval` is the right tool to use here.